examples/island/Island.cpp

```diff
@@ -534,7 +534,8 @@ int main()
                 switch (event->getIf<sf::Event::KeyPressed>()->code)
                 {
                     case sf::Keyboard::Key::Enter:
-                        generateTerrain(terrainStagingBuffer.data());
+                        if (pendingWorkCount == 0 && !bufferUploadPending)
+                            generateTerrain(terrainStagingBuffer.data());
```

Comment:
Does the reading of `pendingWorkCount` not require a lock?

What about clearing the work items and queuing the new changes?